[Qol] Add an option to import/export settings

prateau opened the pull request:
## What are the changes the user will see?
Added an option in the manage data menu to import and export settings

## Why am I making these changes?
This was already started on an older version in this PR and I found some time to update this on the current version.
This is to avoid re-entering manually all my settings when I delete my cookies.

## What are the changes from a developer perspective?
I added two entries in the manage data menu to import and export the settings and reused the already existing import-export function.

I reused the translation from the old PR, I'm only missing japanese.

### Screenshots/Videos
https://github.com/user-attachments/assets/44ad5295-6383-4b66-b5b9-006dca3f33bb


## How to test the changes?
This can be tested by exporting and importing settings with the new option menu.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
   - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

flx-sta commented:
@prateau Thanks for your contribution 
<sub>I know this is your second attempt to get it in, so thanks for the tenacity</sub>

----

### TODO:

- [ ] Resolve the conflicts
- [ ] Add some automated tests coverage for your feature
    - If you need help with writing tests do not hesitate to ask us here and/or in discord #dev-corner

prateau commented:
@flx-sta It's you and the all the team I should thank for working on this game a lot more than I do !

The conflicts are resolved and I added some test on the import/export settings feature. I also refactored this part a tiny bit to allow testing on the import/export functions without having to deal with the ui.

torranx commented:
Hello, sorry for the delayed review and thank you for your contribution. Do you mind updating the PR (again)?

prateau commented:
Hello @torranx updated

DayKev commented:
Functionality looks good, there's a UI issue when importing the data where the text box background disappears.

![image](https://github.com/user-attachments/assets/e83c9d6b-42b3-478b-b841-9a72d5d35098)

DayKev commented:
After importing system data on this branch and then trying to start a new classic run, I get this error:

![image](https://github.com/user-attachments/assets/aa54007d-b499-4c95-bed2-fd21898ddce8)

```
main.ts:13 TypeError: Cannot mix BigInt and other types, use explicit conversions
    at StarterSelectUiHandler.getCurrentDexProps (starter-select-ui-handler.ts:2858:67)
    at StarterSelectUiHandler.updateStarters [as onChange] (starter-select-ui-handler.ts:206:37)
    at DropDown.setCursor (dropdown.ts:314:14)
    at DropDown.resetToDefault (dropdown.ts:441:12)
    at FilterBar.setValsToDefault (filter-bar.ts:153:16)
    at StarterSelectUiHandler.resetFilters (starter-select-ui-handler.ts:949:20)
    at StarterSelectUiHandler.show (starter-select-ui-handler.ts:867:12)
    at doSetMode (ui.ts:445:29)
    at TimerEvent2.callback (ui.ts:452:13)
    at Clock2.update (phaser.js?v=e8cfda40:131135:42)
```

Exporting data appears to work fine though (imported data exported from this branch after switching back to the beta branch and it imported normally).

damocleas commented:
Seems to have conflicts, also commenting to ask if you are still interested in getting this added (considering it has been months and this is round 2 of trying it)

SirzBenjie commented:
Closing since no activity after months. Feel free to reopen if there is interest in getting this in with current codebase.
